script/sql/update_deleted_enlistments_for_deleted_projects.rb

```diff
@@ -0,0 +1,14 @@
+#! /usr/bin/env ruby
+
+raise 'RAILS_ENV is undefined' unless ENV['RAILS_ENV']
+
+require_relative '../config/environment'
+
+# update enlistments deleted field for all deleted projects
+ActiveRecord::Base.connection.execute(
+ "update enlistments set deleted = true
+   from enlistments e
+     inner join projects p on e.project_id = p.id
+     where p.deleted and NOT e.deleted"
+)
```

Comment:
We may need to do more because this will bypass the Edit mechanism.  Something like:

```ruby
editor = Account.find_by_login('ohloh_slave')
Project.find_in_batches(deleted: true) do |projects|
  projects.each do |project|
    project.enlistments.each do |enlistment|
      enlistment.create_edit.undo!(editor) if enlistment.create_edit.allow_undo?
    end
  end
end
```